tensionmap.py

```diff
@@ -147,35 +154,60 @@ def tm_update(obj, context):
 
     # delete the temporary deformed mesh
     object_eval.to_mesh_clear()
-
+    
+    #create vertex color list for faster access
+    number_of_tm_channels = 2
+    vertex_colors = [0]*(number_of_tm_channels*num_vertices)
     # put the new values in the vertex groups
     for i in range(num_vertices):
         add_index = [i]
+        stretch_value = obj.data.tm_minimum
+        squeeze_value = obj.data.tm_minimum
         if weights[i] >= 0:
             # positive: stretched
-            group_squeeze.add(add_index, obj.data.tm_minimum, "REPLACE")
-            group_stretch.add(add_index, max(obj.data.tm_minimum, min(obj.data.tm_maximum, weights[i])), "REPLACE")
+            stretch_value = max(obj.data.tm_minimum, min(obj.data.tm_maximum, weights[i]))
+            if obj.data.tm_enable_vertex_groups:
+                group_squeeze.add(add_index, obj.data.tm_minimum, "REPLACE")
+                group_stretch.add(add_index, stretch_value, "REPLACE")
         else:
             # negative: squeezed
             # invert weights to keep only positive values
-            group_squeeze.add(add_index, max(obj.data.tm_minimum, min(obj.data.tm_maximum, -weights[i])), "REPLACE")
-            group_stretch.add(add_index, obj.data.tm_minimum, "REPLACE")
+            squeeze_value = max(obj.data.tm_minimum, min(obj.data.tm_maximum, -weights[i]))
+            if obj.data.tm_enable_vertex_groups:
+                group_squeeze.add(add_index, squeeze_value, "REPLACE")
+                group_stretch.add(add_index, obj.data.tm_minimum, "REPLACE")
+        vertex_colors[i*number_of_tm_channels] = stretch_value  # red
+        vertex_colors[i*number_of_tm_channels+1] = squeeze_value  # green
 
     if obj.data.tm_enable_vertex_colors:
         colors_tension = get_or_create_vertex_colors(obj, "tm_tension")
         # put the new values from the vertex groups in the vertex colors
         # this is heavy, but vertex colors are stored by vertex loop
         # and there is no simpler way to do it (it would seem)
-        for i in range(len(obj.data.polygons)):
-            polygon = obj.data.polygons[i]
-            for key, value in enumerate(polygon.loop_indices):
-                vertex_color = colors_tension.data[value]
-                vertex = polygon.vertices[key]
-
-                vertex_color.color[0] = group_stretch.weight(vertex)  # red
-                vertex_color.color[1] = group_squeeze.weight(vertex)  # green
-                vertex_color.color[2] = 0.0  # blue
-
+        for poly_idx in range(len(obj.data.polygons)):
+            polygon = obj.data.polygons[poly_idx]
+            for loop_vertex_idx, loop_idx in enumerate(polygon.loop_indices):
+                vertex_color = colors_tension.data[loop_idx]
+                vertex_idx = polygon.vertices[loop_vertex_idx]
+                vertex_color.color[0] = vertex_colors[vertex_idx*number_of_tm_channels]
```

Comment:
I do believe the `# red` and `# green` comments are useful for anyone looking at the code. You could add them back

Reply:
Oh, they were removed by mistake. Re-added!